apps/tools/inventory/fdsnxml2inv/fdsnxml/equipment.cpp

```diff
@@ -271,6 +271,8 @@ Equipment& Equipment::operator=(const Equipment &other) {
 	_installationDate = other._installationDate;
 	_removalDate = other._removalDate;
 	_resourceId = other._resourceId;
+	_calibrationDates = other._calibrationDates;
```

Comment:
Actually child objects are not part of the operator=. That has been ever the contract of this function. Otherwise all other classes should be fixed as well and there might be side-effects when changing semantics. There is no other option than iterating and adding childs from source to target with explicit code.

Reply:
The problem is setSensor(), setDataLogger() and setPreAmplifier here:

https://github.com/SeisComP/main/blob/e7692fff15562063c1b11421b036b2ea3e892f24/apps/tools/inventory/fdsnxml2inv/fdsnxml/channel.h#L153

If the Equipment assignment operator is not changed, then those functions should be changed probably. Otherwise identifiers and calibrations dates will be lost.

Reply:
That is true. It is more of a generic issue that does not exhibit with the core data model. We need to find a generic solution for that.